src/storage/irc_transaction.cpp

```diff
@@ -134,15 +218,54 @@ static rest_api_objects::TableRequirement CreateAssertRefSnapshotIdRequirement(I
 	return req;
 }
 
+rest_api_objects::LoadTableResult IRCTransaction::CommitNewTable(ClientContext &context, const ICTableEntry *table) {
+	auto &ic_catalog = table->catalog.Cast<IRCatalog>();
+	auto table_namespace = table->schema.name;
```

Comment:
Can we move this to `catalog_api.cpp` ?
I'd prefer all the interaction with the REST API to live there.

Also because the `namespace` parameter has to be generated, to support nested namespaces.
The logic for that is in `catalog_api.cpp`

Reply:
done. For now I'm still generating the body in `CommitNewTable` and passing a table entry. Maybe later we can revisit if this is the best place to have it be generated